pjmedia/src/pjmedia/sdp_neg.c

```diff
@@ -1940,15 +1940,18 @@ PJ_DEF(pj_status_t) pjmedia_sdp_neg_cancel_offer(pjmedia_sdp_neg *neg)
                      neg->state == PJMEDIA_SDP_NEG_STATE_REMOTE_OFFER,
                      PJMEDIA_SDPNEG_EINSTATE);
 
-    // No longer needed after #3322.
-    if (0 && neg->state == PJMEDIA_SDP_NEG_STATE_LOCAL_OFFER &&
+    if (neg->state == PJMEDIA_SDP_NEG_STATE_LOCAL_OFFER &&
         neg->active_local_sdp) 
     {
         /* Increment next version number. This happens if for example
          * the reinvite offer is rejected by 488. If we don't increment
          * the version here, the next offer will have the same version.
          */
-        neg->active_local_sdp->origin.version++;
+        // No longer needed after #3322.
+        // neg->active_local_sdp->origin.version++;
```

Comment:
With the code is commented, the next local SDP will still have version incremented, right?

Reply:
Only if it's different

Reply:
Different to `active_local_sdp` or the last rejected?
(I guess the comparison has to be done against any last SDP, in this case it is the rejected one?)

Reply:
With the change in #3393 that no longer increments the version, the current behavior is as follows:
SDP version 1, reoffer SDP v 2, rejected, compare with `neg->last_sent`, if same send SDP version 1 else version 2.

Without the patch, the comparison will be done to SDP in the rejected reoffer.
With the patch, the comparison will be done to `active_local_sdp`.

Refer to the PR desc of #3393.

Note: we have a Sipp script to test the scenario in `uas-reinv-after-failed-nego.xml`.

Reply:
Okay, so the comparison for SDP version increment is always done against `active_local_sdp` (any rejected offer is ignored).

Re: Sipp script, it does not seem to verify the SDP version, just reproducing the scenario, right?

Reply:
Correct, any rejected offer should be ignored (as the RFC says: `as if no re-INVITE/UPDATE had been issued`).
It may cause the SDP version to decrease in the scenario: `SDP version 1, reoffer SDP v 2, rejected, if we send reoffer with the same SDP, the version will be back to 1`, which may seem strange, but I suppose that's what the RFC recommends.

Yes, for reproducing only.